tutorials/backend/hasura/tutorial-site/content/custom-business-logic/1-actions.md

```diff
@@ -0,0 +1,155 @@
+
+```
+type Query {
+  auth0 : auth0_profile
+}
+```
+
+### Types definition
+
+We defined that the response type of the action is `auth0_profile`. So what do we want in return from the Auth0 API? We want the `id`, `email` and `picture` fields which aren't stored on our database so far.
+
+```
+type auth0_profile {
+  id : String
+  email : String
+  picture : String
+}
+```
+
+All three fields are of type String. Note that `auth0_profile` is an object type which has 3 keys (id, email and picture) and we are returning this object in our response.
+
+We will change the Handler URL later once we write our REST API and deploy it on a public endpoint.
+
+![Create Action](https://storage.googleapis.com/graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura/create-action.png)
+
+Click on Create once you are done configuring the above fields.
+
+## Write a REST API
+
+Now that the Action has been created, let's write a REST API in a simple Node.js Express app that can later be configured for this Action. 
+
+Head to the `Codegen` tab to quickly get started with boilerplate code :) 
+
+Click on `Try on Glitch` to deploy a server. Glitch is a platform to build and deploy apps (Node.js) and is a quick way to test and iterate code on the cloud.
+
+![Action codegen](https://storage.googleapis.com/graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura/action-codegen-tab.png)
+
+Now replace the contents of `src/server.js` with the following:
+
+```javascript
+const express = require("express");
+const bodyParser = require("body-parser");
+const fetch = require('node-fetch');
+
+const app = express();
+
+const PORT = process.env.PORT || 3000;
+
+app.use(bodyParser.json());
+
+const getProfileInfo = (user_id) => {
+    const headers = {'Authorization': 'Bearer '+process.env.AUTH0_MANAGEMENT_API_TOKEN};
+    console.log(headers);
+    return fetch('https://' + process.env.AUTH0_DOMAIN + '/api/v2/users/'+user_id,{ headers: headers})
+        .then(response => response.json())
+}
+
+app.post('/auth0', async (req, res) => {
+
+  // get request input
+  const { session_variables } = req.body;
+  
+  const user_id = session_variables['x-hasura-user-id'];
+  // make a rest api call to auth0
+  return getProfileInfo(user_id).then( function(resp) {
+    console.log(resp);
+    if (!resp) {
+      return res.status(400).json({
+        message: "error happened"
+      })
+    }
+    return res.json({
+      email: resp.email,
+      picture: resp.picture
+    })
+  });
+
+});
+
+app.listen(PORT);
+
+
+```
+
+In the server above, let's breakdown what's happening:
+
+- We receive the payload `session_variables` as the request body from the Action.
+- We make a request to the [Auth0's Management API](https://auth0.com/docs/api/management/v2/create-m2m-app), passing in the user_id to get details about this user.
+- Once we get a response, we return back the object `{email: resp.email, picture: resp.picture}` as response. Else, we return an error case.
```

Comment:
This sentence can be a bit confusing: `Once we get a response, we return back the object as a response`. Maybe specify where we get the first response from and where we return the object as a response.